pkg/harvester/edit/network.harvesterhci.io.vlanconfig/index.vue

```diff
@@ -0,0 +1,363 @@
+
+      const out = clusterNetworks.map((n) => {
+        return {
+          label: n.id,
+          value: n.id,
+        };
+      });
+
+      return [{
+        label: this.t('harvester.network.clusterNetwork.create'),
+        value: 'empty',
+      }, {
+        label:    'divider',
+        disabled: true,
+        kind:     'divider'
+      }, ...out];
+    },
+
+    nodeOptions() {
+      const inStore = this.$store.getters['currentProduct'].inStore;
+      const nodes = this.$store.getters[`${ inStore }/all`](NODE);
+
+      return nodes.filter(n => !n.isUnSchedulable).map((node) => {
+        return {
+          label: node.nameDisplay,
+          value: node.id
+        };
+      });
+    },
+
+    mtu: {
+      get() {
+        return this.value?.spec?.uplink?.linkAttributes?.mtu;
+      },
+
+      set(value) {
+        set(this.value, 'spec.uplink.linkAttributes.mtu', value);
+      }
+    },
+
+    bondOptionMode: {
+      get() {
+        return this.value?.spec?.uplink?.bondOptions?.mode;
+      },
+
+      set(value) {
+        set(this.value, 'spec.uplink.bondOptions.mode', value);
+      },
+    },
+
+    miimon: {
+      get() {
+        return this.value?.spec?.uplink?.bondOptions?.miimon;
+      },
+
+      set(value) {
+        set(this.value, 'spec.uplink.bondOptions.miimon', value);
+      },
+    },
+
+    // TODO: It's a temporary modifications, for now only support VLAN.
+    // Will be support SR-IOV in the future and feel free change the design
+    typeOptions() {
+      return [{
+        label: 'VLAN',
+        value: 'vlan',
+      }];
+    },
+
+    bondOptions() {
+      return [
+        'balance-rr',
+        'active-backup',
+        'balance-xor',
+        'broadcast',
+        '802.3ad',
+        'balance-tlb',
+        'balance-alb',
+      ];
+    },
+  },
+
+  methods: {
+    selectClusterNetwork(e) {
+      if (!e || e.value === 'empty') {
+        this.createClusterNetwork = true;
+        this.value.spec.clusterNetwork = '';
+
+        Vue.nextTick(() => this.$refs.clusterNetwork.focus());
+      } else {
+        this.createClusterNetwork = false;
+      }
+    },
+
+    async saveClusterNetwork() {
+      const clusterNetwork = this.value.spec.clusterNetwork;
+      const inStore = this.$store.getters['currentProduct'].inStore;
+
+      const found = this.$store.getters[`${ inStore }/byId`](HCI.CLUSTER_NETWORK, clusterNetwork);
+
+      if (!found) {
+        const newResource = await this.$store.dispatch(`${ inStore }/create`, {
+          metadata: { name: clusterNetwork },
+          type:     HCI.CLUSTER_NETWORK,
+        });
+
+        return await newResource.save();
+      } else {
+        return Promise.resolve();
+      }
+    },
+
+    validate() {
+      const errors = [];
+
+      const nics = this.value.spec?.uplink?.nics || [];
+      const nicRequired = this.t('validation.required', { key: this.t('harvester.vlanConfig.uplink.nics.label') }, true);
```

Comment:
This produces the error message `NICs is required.` I think it needs a different validation message:
` 'At least {count} {key} {count, plural, =1 {is} other {are}} required'`  (where in this case key = "NIC" count = 1)